mail_edit/static/src/js/mail_edit.js

```diff
@@ -0,0 +1,73 @@
+/* © 2014-2017 Sunflower IT <www.sunflowerweb.nl>*/
+
+"use strict";
+openerp.mail_edit = function (instance) {
+    var _t = instance.web._t;
+    instance.mail.ThreadMessage.include({
+        bind_events: function () {
+            this._super.apply(this, arguments);
+            this.$('.oe_edit').on('click', this.on_message_edit);
+        },
+
+        on_message_edit: function () {
+            // Generate email subject as possible from record_name and subject
+            var subject = [];
+            if (this.record_name && (this.show_record_name ||
+                                     this.parent_id))
+            {
+                subject.push(this.record_name);
+            }
+            if (this.subject) {
+                subject.push(this.subject);
+            } else if (subject.length < 2) {
+                subject.push(_t("(No subject)"));
+            }
+
+            // Get only ID from the attachments
+            var attachment_ids = [];
+            for (var n in this.attachment_ids) {
+                attachment_ids.push(this.attachment_ids[n].id);
+            }
+
+            // Get necessary fields from the forwarded message
+            var header = [
+                _t("From: ") + this.author_id[1],
+                _t("Date: ") + this.date,
+            ];
+            if (this.subject) {
+                header.push(_t("Subject: ") + this.subject);
+            }
+            if (this.email_to) {
+                header.push(_t("To: ") + this.email_to);
+            }
+            if (this.email_cc) {
+                header.push(_t("CC: ") + this.email_cc);
+            }
+            header = header.map(_.str.escapeHTML).join("<br/>")
+
+            var context = {
+                default_attachment_ids: attachment_ids,
+                default_body: this.body,
+                default_model: this.model,
+                default_res_id: this.res_id,
+                default_subject: subject.join(": "),
+                default_message_id: this.id,
+            };
+
+            if (this.model && this.res_id) {
+                context.default_destination_object_id =
+                    [this.model, this.res_id].join();
+            }
+
+            // Get the action data and execute it to open the composer wizard
+            var do_action = this.do_action;
```

Comment:
the canonical (and that's only odoo canonical, by no way js canonical) way is to assign `this` to `self` and then call stuff on `self` in handlers, but that's totally arbitrary and what you do is perfectly fine